tck/src/test/java/org/reactivestreams/tck/PublisherVerificationTest.java

```diff
@@ -717,6 +735,79 @@ public void cancel() {
   }
 
   /**
+   * Verification using a Publisher that supports multiple subscribers
+   * @param shouldBlowUp if true {@link RuntimeException} will be thrown during second subscription.
+   */
+  final PublisherVerification<Integer> multiSubscribersPublisherVerification(final boolean shouldBlowUp) {
+    return new PublisherVerification<Integer>(newTestEnvironment()) {
+
+      @Override
+      public Publisher<Integer> createPublisher(final long elements) {
+        return new Publisher<Integer>() {
+
+          private final Collection<Demand> demands = new CopyOnWriteArrayList<Demand>();
+          private final AtomicLong source = new AtomicLong(elements);
+
+          @Override
+          public void subscribe(Subscriber<? super Integer> s) {
+            if (shouldBlowUp && !demands.isEmpty()) {
+              // onSubscribe first
+              new Demand(s);
+              s.onError(new RuntimeException("Unexpected additional subscriber"));
+            } else {
+              demands.add(new Demand(s));
+            }
+          }
+
+          class Demand implements Subscription {
+
+            final AtomicBoolean canceled = new AtomicBoolean();
+            Subscriber<? super Integer> subscriber;
+
+            Demand(Subscriber<? super Integer> subscriber) {
+              this.subscriber = subscriber;
+              this.subscriber.onSubscribe(this);
```

Comment:
Not a big fan of letting `this` escape before constructor has finished running - subscriber can immediately synchronously call things on us right away. I think we're pretty safe with current impl but generally would like to avoid this pattern, WDYT?